pom.xml

```diff
@@ -254,7 +271,7 @@ limitations under the License.
                             <exclude>release.properties</exclude>
                             <exclude>**/*.cq4</exclude>
                             <exclude>**/directory-listing.cq4t</exclude>
-                            <exclude>pmd-rules.xml</exclude>
+                            <exclude>${pmd-config-url}</exclude>
```

Comment:
Nit:

Isn't it better to exclude pmd-rules.xml directly?

Reply:
This was the best solution I could find that allow you to do the pmd check even if your executing mvn within one of the module directories. Not very beautiful, but each module can redefine the variable and get it work.

This was the most elegant solution I could find that:
- execute pmd as part of compile stage
- allow "mvn compile" to work both at top level and at module level
- don't "break" other maven plug-ins in the directory

Played around with a dedicated build-tool module, but could get that to work with point 3 in the list. Could perhaps be worked-around with a dedicated compile module as well that would act as parent to all the other "real" modules. But that's just to much complexity.

If only the pmd-plugin would allow me to specify the rules directly in the pom.xml...